### PR TITLE
constants'n'enums

### PR DESCRIPTION
**Pull request by zdimension**

### CODE REVIEW

File: GAME/LARAFIRE.C [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #include "GAMEFLOW.H"
 #include "SPECIFIC.H"
 #include "OBJECTS.H"
+#include "LARA.H"
```
Comment (Gh0stBlade) [inline GAME/LARAFIRE.C]: Includes must be ordered alphabetically.

File: GAME/LARA.C [hunk 2]
```diff
@@ -82,19 +82,76 @@ void lara_as_trpose(struct ITEM_INFO* item, struct COLL_INFO* coll)//1CDE8(<), 1
 	}
 }
 
-void GetTighRopeFallOff(long Regularity)//1CD28, 1CEBC
+void GetTighRopeFallOff(long Regularity)//1CD28, 1CEBC (f)
```
Comment (Gh0stBlade) [inline GAME/LARA.C]: Uppercase F is required.

File: GAME/CONTROL.C [hunk 3]
```diff
@@ -34,6 +34,7 @@
 #include "TOMB4FX.H"
 
 #include <assert.h>
+#include "LOT.H"
```
Comment (Gh0stBlade) [inline GAME/CONTROL.C]: Non c library includes should be grouped together.

File: GAME/CONTROL.C [hunk 4]
```diff
@@ -1045,9 +1046,35 @@ void RemoveRoomFlipItems(struct room_info* r)//1F938(<), 1FB4C(<) (F)
 	}
 }
 
-void FlipMap(int FlipNumber)
+void FlipMap(int FlipNumber) // (F)
 {
-	S_Warn("[FlipMap] - Unimplemented!\n");
+	struct room_info* r = room;
+	for(int i = 0; i < number_rooms; i++, r++)
```
Comment (Gh0stBlade) [inline GAME/CONTROL.C]: i must be declared at the very beginning.

File: GAME/CONTROL.C [hunk 5]
```diff
@@ -1045,9 +1046,35 @@ void RemoveRoomFlipItems(struct room_info* r)//1F938(<), 1FB4C(<) (F)
...
+	{
+		if (r->flipped_room >= 0 && r->FlipNumber == FlipNumber)
+		{
+			struct room_info temp;
```
Comment (Gh0stBlade) [inline GAME/CONTROL.C]: Must be declared at the very beginning of method.

File: GAME/CONTROL.C [hunk 6]
```diff
@@ -1045,9 +1046,35 @@ void RemoveRoomFlipItems(struct room_info* r)//1F938(<), 1FB4C(<) (F)
...
+			flipped->flipped_room = -1;
+			r->item_number = flipped->item_number;
+			r->fx_number = flipped->fx_number;
+			AddRoomFlipItems(r);
+		}
+	}
+	flip_stats[FlipNumber] = flip_stats[FlipNumber] == 0;
+	flip_status = flip_stats[FlipNumber] == 0;
+	{
+		struct creature_info* cinfo = baddie_slots;
```
Comment (Gh0stBlade) [inline GAME/CONTROL.C]: Must be declared at the very beginning of method.

File: GAME/CONTROL.C [hunk 7]
```diff
@@ -1045,9 +1046,35 @@ void RemoveRoomFlipItems(struct room_info* r)//1F938(<), 1FB4C(<) (F)
...
+		for (int slot = 0; slot < 6; slot++, cinfo++)
```
Comment (Gh0stBlade) [inline GAME/CONTROL.C]: Must be declared at the very beginning of method.

**Comment by Gh0stBlade** [comment]
Very good changes. About OSX automated building. Yes, travis supports it, I'm not experienced with that operating system unfortunately. Would be cool to see it added some time despite the revert.

File: GAME/LARAFIRE.C (hunk 1 above)
Comment (zdimension) [reply GAME/LARAFIRE.C]: fixed

File: GAME/LARA.C (hunk 2 above)
Comment (zdimension) [reply GAME/LARA.C]: fixed

File: GAME/CONTROL.C (hunk 3 above)
Comment (zdimension) [reply GAME/CONTROL.C]: fixed

File: GAME/CONTROL.C (hunk 4 above)
Comment (zdimension) [reply GAME/CONTROL.C]: fixed

File: GAME/CONTROL.C (hunk 5 above)
Comment (zdimension) [reply GAME/CONTROL.C]: c89 only wants at the beginning of scope which is the case here
\+ tested with gcc -pedantic

File: GAME/CONTROL.C (hunk 6 above)
Comment (zdimension) [reply GAME/CONTROL.C]: c89 only wants at the beginning of scope (which is why I used the { } brackets here, to create a new block scope)

File: GAME/CONTROL.C (hunk 7 above)
Comment (zdimension) [reply GAME/CONTROL.C]: c89 only wants at the beginning of scope (which is why I used the { } brackets here, to create a new block scope)

**Pull request merged**